pkg/api/filter.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+// Filter return true iff specific api needs to be filtered/ removed from list
+type Filter func(api Api) bool
+
+// Filter apply all passed filters and return new filtered array
+func (m ApiMap) Filter(filters ...Filter) ApiMap {
+	apis := make(ApiMap)
```

Comment:
nit/question: indentation looks kinda off in this file? like it's a level too deep. 
Did you run `make format`?

Reply:
it is done now